crates/datasources/src/json/stream.rs

```diff
@@ -0,0 +1,82 @@
+use std::pin::Pin;
+use std::sync::Arc;
+use std::sync::Mutex;
+use std::task::{Context, Poll};
+
+use datafusion::arrow::datatypes::{Schema, SchemaRef};
+use datafusion::arrow::json::ReaderBuilder;
+use datafusion::arrow::record_batch::RecordBatch;
+use datafusion::error::DataFusionError;
+use datafusion::execution::TaskContext;
+use datafusion::physical_plan::streaming::PartitionStream;
+use datafusion::physical_plan::{RecordBatchStream, SendableRecordBatchStream};
+use futures::Stream;
+use futures::StreamExt;
+use serde_json::{Map, Value};
+
+pub struct JsonStream {
+    schema: Arc<Schema>,
+    stream: Pin<Box<dyn Stream<Item = Result<RecordBatch, DataFusionError>> + Send>>,
```

Comment:
Datafusion has a `SendableRecordBatchStream` alias for this which is usually easier to write.

Reply:
much to my great displeasure, `SendableRecordBatchStream` is this just record batches, no results.